docs/internals/optimizer.rst

```diff
@@ -958,7 +959,7 @@ DeadCodeEliminator
 This optimization stage removes unreachable code.
 
 Unreachable code is any code within a block which is preceded by a
-leave, return, invalid, break, continue, selfdestruct or revert.
+leave, return, invalid, break, continue, selfdestruct, revert or by a call to a user-defined function that infinitely recurses.
```

Comment:
```suggestion
leave, return, invalid, break, continue, selfdestruct, revert or by a call to a user-defined function that recurses infinitely.
```
I may have suggested this in the first place, but the other way round is more idiomatic, isn't it? Not too important in any case (in the end it's a rather silly special case and technicality anyways).

Reply:
If it's a special case, I suggest even more special wording - `ad infinitum`.
Jokes aside, yeah, your suggestion does sound better.